### PR TITLE
Doc change. translation to Russian some array functions

### PR DESCRIPTION
**Pull request by den-crane**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
* Documentation

### CODE REVIEW

File: docs/ru/query_language/functions/array_functions.md [hunk 1]
```diff
@@ -658,4 +658,115 @@ SELECT arrayReverseSort((x, y) -> -y, [4, 3, 5], [1, 2, 3]) AS res;
 
 Особенная функция. Смотрите раздел ["Функция arrayJoin"](array_join.md#functions_arrayjoin).
 
+## arrayDifference(arr)
+
+Принимает массив, возвращает массив разностей между парами соседних элементов.
```
Comment (alexey-milovidov) [inline docs/ru/query_language/functions/array_functions.md]: Not descriptive enough. In what order the difference is applied?
I can read it as:
`[1 - 2, 3 - 4] = [-1, -1]`.

Comment (alexey-milovidov) [reply docs/ru/query_language/functions/array_functions.md]: Probably we should mind about the return type and the corner cases:

`SELECT arrayDifference([0, 10000000000000000000])`

File: docs/ru/query_language/functions/array_functions.md [hunk 2]
```diff
@@ -658,4 +658,115 @@ SELECT arrayReverseSort((x, y) -> -y, [4, 3, 5], [1, 2, 3]) AS res;
...
+
+```text
+┌─arrayEnumerateDense([10, 20, 10, 30])─┐
+│ [1,2,1,3]                             │
+└───────────────────────────────────────┘
+```
+
+## arrayIntersect(arr)
+
+Принимает массивы, возвращает пересечение элементов массивов.
```
Comment (alexey-milovidov) [inline docs/ru/query_language/functions/array_functions.md]: Ok but what about the return type?

File: docs/ru/query_language/functions/array_functions.md [hunk 3]
```diff
@@ -658,4 +658,115 @@ SELECT arrayReverseSort((x, y) -> -y, [4, 3, 5], [1, 2, 3]) AS res;
...
+
+```text
+┌─no_intersect─┬─intersect─┐
+│ []           │ [1]       │
+└──────────────┴───────────┘
+```
+
+## arrayReduce(agg_func, arr1, ...)
+
+Применяет агрегатную функцию к массиву и возвращает ее результат.
```
Comment (alexey-milovidov) [inline docs/ru/query_language/functions/array_functions.md]: Should note that the aggregate function name is passed as a constant string literal.
And about the way how parametric aggregate functions can be specified.

File: docs/ru/query_language/functions/array_functions.md [hunk 4]
```diff
@@ -658,4 +658,115 @@ SELECT arrayReverseSort((x, y) -> -y, [4, 3, 5], [1, 2, 3]) AS res;
...
+
+```text
+┌─arrayReduce('maxIf', [3, 5], [1, 0])─┐
+│                                    3 │
+└──────────────────────────────────────┘
+```
+
+## arrayReverse(arr)
+
+Возвращает массив того же размера, что и исходный массив, содержащий элементы в обратном порядке.
```
Comment (alexey-milovidov) [inline docs/ru/query_language/functions/array_functions.md]: Should note that the `reverse` function has overload for arrays.

File: docs/ru/query_language/functions/array_functions.md [hunk 5]
```diff
@@ -647,15 +647,156 @@ SELECT arrayReverseSort((x, y) -> -y, [4, 3, 5], [1, 2, 3]) AS res;
...
 
-## arrayJoin(arr)
+## arrayJoin(arr) {#array_functions-arrayjoin}
 
 Особенная функция. Смотрите раздел ["Функция arrayJoin"](array_join.md#functions_arrayjoin).
 
+## arrayDifference(arr) {#array_functions-arraydifference}
+
+Принимает массив, возвращает массив разностей между соседними элементами. Первым элементом будет 0, вторым разность между вторым и первым элементами исходного массива, и т.д.
+Результирующий массив имеет тип Array(Int64) для целых чисел и Array(Float64) для чисел с плавающей точкой. Другие типы, в том числе Decimal, не поддерживаются.
```
Comment (alexey-milovidov) [inline docs/ru/query_language/functions/array_functions.md]: That's incorrect:
```
SELECT toTypeName(arrayDifference([1, 1]))
```

Comment (den-crane) [reply docs/ru/query_language/functions/array_functions.md]: > That's incorrect:
> 

How about: Тип элементов результирующего массива определяется правилами выведения типов при вычитании (напр. UInt8 - UInt8 = Int16).

Comment (alexey-milovidov) [reply docs/ru/query_language/functions/array_functions.md]: Да, это соответствует на 100%!

**Pull request merged**